plugins/nvme/fu-nvme-device.c

```diff
@@ -75,63 +75,42 @@ fu_nvme_device_get_guid_safe(const guint8 *buf, guint16 addr_start)
 				    FWUPD_GUID_FLAG_MIXED_ENDIAN);
 }
 
+static gboolean
+fu_nvme_device_ioctl_buffer_cb(FuIoctl *self,
+			       gpointer wrapped,
+			       guint8 *buf,
+			       gsize bufsz,
+			       GError **error)
+{
+	struct nvme_admin_cmd *cmd = (struct nvme_admin_cmd *)wrapped;
+	memcpy(&cmd->addr, &buf, sizeof(gpointer)); /* nocheck:blocked */
```

Comment:
I don't think we should be in the habit of introducing code that is `nocheck:blocked` in the plugins.  I'd say use the wrapper here even if we know it's perfectly safe.

Reply:
Ack, agree. It's not pretty, but it's an ioctl...